task4/func.js

```diff
@@ -0,0 +1,17 @@
+function Person(name,age) {
+    this.name = name;
+    this.age = age;
+}
+
+Person.prototype.introduce = function () {
+    return `My name is ${this.name} and I am ${this.age}`;
+}
+
+function myNew(constructorName, ...args) {
+    const objPrototype = Object.create(constructorName.prototype);
+    const instance = constructorName.apply(objPrototype, args);
+    return (typeof instance === 'object' && instance) || objPrototype;
```

Comment:
It won't work for functions, because functions have another type via executing typeof
Please, add this checking:
`((typeof instance === 'object' || typeof instance === 'function') && instance) || objPrototype`